File: iOS/TutorFind/Grades/Grade.swift
import Foundation
import Firebase
struct Grade: Codable, DictionaryCodable, StorableOnFirebase, Hashable{

	var id = ""
	var studentID = ""
	var professorID = ""
	var courseID = ""
	var studentName = ""
	var professorName = ""
	var grade = ""
	var title = ""
	var comments = ""
	var courseName = ""
	var date = Date()
	static var basePathString: String = "grades"
	static var genericName: String = "Βαθμολογία"
	init(){}
	
	init?(id: String, dict: [String: Any]){
		guard
				let studentID = dict["studentID"] as? String
				,let professorID = dict["professorID"] as? String
				,let courseID = dict["courseID"] as? String
				,let studentName = dict["studentName"] as? String
				,let professorName = dict["professorName"] as? String
				,let grade = dict["grade"] as? String
				,let title = dict["title"] as? String
				,let comments = dict["comments"] as? String
				,let dateString = dict["date"] as? String
					,let courseName = dict["courseName"] as? String
				,let date = Date.fromString(string: dateString)
		else{return nil}
		self.id=id
		self.studentID=studentID
		self.courseID=courseID
		self.professorID=professorID
		self.studentName=studentName
		self.professorName=professorName
		self.grade=grade
		self.title=title
		self.comments=comments
		self.date=date
		self.courseName=courseName
	}
	func toDict()->[String: Any]{
		var dict = [String:Any]()
		dict["studentID"] = studentID
		dict["professorID"] = professorID
		dict["studentName"] = studentName
		dict["professorName"] = professorName
		dict["grade"] = grade
		dict["title"] = title
		dict["courseID"] = courseID
		dict["comments"] = comments
		dict["date"]=date.toString()
		dict["courseName"]=courseName
		return dict
	}
}

File: Android/app/src/main/java/com/team10/tutorfind/models/Grade.java
package com.team10.tutorfind.models;

import java.util.HashMap;
import java.util.Map;

public class Grade {
//    var id = ""
//    var studentID = ""
//    var professorID = ""
//    var studentName = ""
//    var professorName = ""
//    var grade: Double = 0.0
//    var title = ""
//    var comments = ""
//    var courseName = ""
//    var date = Date() 22-11-2021 09:59

    String id="";
    String studentID="";
    String professorID="";
    String studentName="";
    String professorName="";
    float grade=0;
    String title="";
    String comments="";
    String courseName="";
    String date="";

    public Map<String, Object> toMap() {
        HashMap<String, Object> result = new HashMap<>();

        result.put("id", id);
        result.put("studentID", studentID);
        result.put("professorID", professorID);
        result.put("studentName", studentName);
        result.put("professorName", professorName);
        result.put("grade", grade);
        result.put("title", title);
        result.put("comments", comments);
        result.put("courseName", courseName);
        result.put("date", date);

        return result;
    }

    public Grade fromMap(Map<String, Object> m) {
        String id = (String) m.get("id");
        String studentID = (String) m.get("studentID");
        String professorID = (String) m.get("professorID");
        String studentName = (String) m.get("studentName");
        String professorName = (String) m.get("professorName");
        float grade = (float) m.get("grade");
        String title = (String) m.get("title");
        String comments = (String) m.get("comments");
        String courseN = (String) m.get("courseName");
        String date = (String) m.get("date");

        if (id == null || studentID == null || professorID == null || studentName == null || professorName == null || title == null || comments == null || courseN == null || date == null)
            return null;
        else {
            this.id = id;
            this.studentID = studentID;
            this.professorID = professorID;
            this.studentName = studentName;
            this.professorName = professorName;
            this.grade = grade;
            this.title = title;
            this.comments = comments;
            this.courseName = courseN;
            this.date = date;
        }

        return this;
    }
}
